Userland/Applications/Browser/Tab.cpp

```diff
@@ -47,6 +55,15 @@ URL url_from_user_input(const String& input)
     StringBuilder builder;
     builder.append("http://");
     builder.append(input);
+    url = URL(builder.build());
+
+    if ((url.is_valid() && (may_url_exist(input))) || s_search_engine_format == "{}")
```

Comment:
```suggestion
    if ((url.is_valid() && (may_url_exist(input))) || s_search_engine_format == "{}"sv)
```

Not really important, but let's not `strlen()` all the time : )

Reply:
> The name `may_url_exist` sounds a bit weird, but I'm not entirely sure what it's supposed to do.

It checks if URL is likely to exist - there are no URLs with spaces, or without dots (except `localhost`). Invalid URLs are also treated as potential search queries. I didn't have a better idea for the name.
The next step would be checking if domain is valid (e.g. it has valid suffix).

Reply:
I think it's fine, it just sounds odd is all.